src/waveform/renderers/waveformrenderbeat.cpp

```diff
@@ -22,24 +22,38 @@ WaveformRenderBeat::~WaveformRenderBeat() {
 void WaveformRenderBeat::setup(const QDomNode& node, const SkinContext& context) {
     m_beatColor.setNamedColor(context.selectString(node, "BeatColor"));
     m_beatColor = WSkinColor::getCorrectColor(m_beatColor).toRgb();
+    m_barColor.setNamedColor(context.selectString(node, "BarColor"));
+    m_barColor = WSkinColor::getCorrectColor(m_barColor).toRgb();
+    m_phraseColor.setNamedColor(context.selectString(node, "PhraseColor"));
+    m_phraseColor = WSkinColor::getCorrectColor(m_phraseColor).toRgb();
 }
 
 void WaveformRenderBeat::draw(QPainter* painter, QPaintEvent* /*event*/) {
     TrackPointer trackInfo = m_waveformRenderer->getTrackInfo();
 
+    // No track, nothing to do
     if (!trackInfo)
         return;
 
+    // Gets beats from the track
     BeatsPointer trackBeats = trackInfo->getBeats();
+    // No beats, nothing to do
     if (!trackBeats)
         return;
 
+    // Set up Alpha values according to the renderer
     int alpha = m_waveformRenderer->beatGridAlpha();
+    // Not visible, nothing to do
     if (alpha == 0)
         return;
     m_beatColor.setAlphaF(alpha/100.0);
+    m_barColor.setAlphaF(alpha/100.0);
+    m_phraseColor.setAlphaF(alpha/100.0);
 
+    // Get number of sample in the track
```

Comment:
Your commentis an indicator for a need to rename the getTrackSamples to getNumberOfSamples() or something.

Reply:
Humm. I agree on that statement, but I was not feeling comfortable modifying a different class.

I will look at it.

Reply:
Done